src/app/ReadHandler.h

```diff
@@ -416,9 +427,16 @@ class ReadHandler : public Messaging::ExchangeDelegate
 
     void PersistSubscription();
 
-    // Helpers for managing our state flags properly.
-    void SetStateFlag(ReadHandlerFlags aFlag, bool aValue = true);
-    void ClearStateFlag(ReadHandlerFlags aFlag);
+    /// @brief This function modifies a state Flag in the read handler and schedules an engine run if the read handler want from a
+    /// non reportable state to a reportable state, resulting in the emission of a report.
+    /// @param aFlag Flag to set
+    /// @param aValue Flag new value
+    void SetStateFlagAndScheduleReport(ReadHandlerFlags aFlag, bool aValue = true);
```

Comment:
The thing is, it might not schedule a report run...  Is the goal here to make it clear from the function name that it can schedule a report run and you don't have to do it manually?  Describing this sort of complicated behavior in function names is tough; I would be tempted to leave it as SetStateFlag and document the report bits.